src/Tools/dotnet-monitor/CollectionRules/CollectionRulePipeline.cs

```diff
@@ -241,5 +249,24 @@ private void InvokeStartCallback()
                 _invokedStartCallback = true;
             }
         }
+
+        public CollectionRulePipelineState GetPipelineState()
+        {
+            CollectionRulePipelineState pipelineStateCopy = new(_stateHolder.ActionCountLimit, _stateHolder.ActionCountSlidingWindowDuration, _stateHolder.RuleDuration);
```

Comment:
Another consideration here:
- Instead of having a copied object that is the same type (CollectionRulePipelineState), create a new object type that has all the necessary data (deep copied) but none of the locking since it won't be required.